app/models/nic/base.rb

```diff
@@ -22,7 +22,7 @@ class Base < ActiveRecord::Base
     validates :mac, :uniqueness => {:scope => :virtual},
               :if => Proc.new { |nic| nic.managed? && nic.host && nic.host.managed? && !nic.host.compute? && !nic.virtual? }, :allow_blank => true
```

Comment:
what about introcuding `actively_managed?` method:

```ruby
def actively_managed?
  SETTINGS[:unattended] && nic.managed? && nic.host && nic.host.managed?
end
```

This code repeats a lot in the code to deserve some common place (bonus points for desciptive commentary about why we can't use the `managed?` flag directly: I know just because I've read https://github.com/theforeman/foreman/pull/2338#issuecomment-99818385 but I'm sure I will forget in 3 days)